### ISSUE TITLE
Linebreaks being removed from pre tags (regression from 1.5.3?)

### ISSUE THREAD

**Issue by cyotek** [xoofx/NUglify#30]
Hello,

It appears that white-space is being removed from `pre` tags even though it is pretty essential it remains. The default `HtmlSettings`  suggests that `pre` and `textarea` tags should be excluded from white-space removal. However, in the current 1.5.6, this seems to be completely broken no matter what settings I use (unless I set `CollapseWhitespaces` to `false`, which isn't really a good workaround for minification)

I create a small test project with the following test modelled after the existing NUglify tests

```c#
    [Test]
    public void PreTagRetainsWhitespace()
    {
      HtmlSettings settings;

      settings = new HtmlSettings();
      //settings.CollapseWhitespaces = false;

      equal(minify("<pre>Line1\nLine2</pre>", settings), "<pre>Line1\nLine2</pre>");
      equal(minify("<pre>    Line1\n    Line2</pre>", settings), "<pre>    Line1\n    Line2</pre>");
      equal(minify("<pre><code>Line1\nLine2</code></pre>", settings), "<pre><code>Line1\nLine2</code></pre>");
      equal(minify("<pre><code>    Line1\n    Line2</code></pre>", settings), "<pre><code>    Line1\n    Line2</code></pre>");
    }
```

Using 1.5.6, the first test fails - the line-breaks have been incorrectly removed.

    <pre>Line1 Line2</pre>

Dropping NUglify versions sequentially, I found that when using 1.5.3 the first test passes, but the second one fails - the leading white-space on the first line has been removed. I tested the same code snippet in Edge and the white-space is retained so I think this is probably the wrong behaviour in NUgily (although I dimly remember issues in distant past with leading white-space on textareas so this may need looking into further, although certainly in the case of `pre` I think it should be kept)

```
<pre>Line1
    Line2</pre>
```

I had a look at the source code to see if there was anything obvious, my guess is 91ce12d5e0f5b3fb2bfca4cff46db464d9124e5b introduced it. Unfortunately I'm not even remotely familiar with the code base or how it works in order to propose a possible fix. On the plus side, at least I know I can revert back to older version as a temporary workaround.

Hope this helps, let me know if you need any more information.

Thanks;
Richard Moss

**Comment by xoofx** [comment]
Thanks for the report, indeed, this was a regression (and no test about this 😓 ) but it should be fixed by 1e8e809

**Comment by cyotek** [comment]
Tested 1.5.7 and everything looks to be fine. Thanks for the quick turnaround!
